fsspec/caching.py

```diff
@@ -406,23 +445,33 @@ def _fetch(self, start, end):
                 self.cache = self.cache[self.blocksize * num :]
         return out
 
-    def __len__(self):
+    def __len__(self) -> int:
         return len(self.cache)
 
 
 class AllBytes(BaseCache):
     """Cache entire contents of the file"""
 
-    name = "all"
+    name: ClassVar[str] = "all"
 
-    def __init__(self, blocksize=None, fetcher=None, size=None, data=None):
-        super().__init__(blocksize, fetcher, size)
+    def __init__(
+        self,
+        blocksize: int | None = None,
+        fetcher: Fetcher | None = None,
+        size: int | None = None,
+        data: bytes | None = None,
+    ) -> None:
+        super().__init__(blocksize, fetcher, size)  # type: ignore[arg-type]
         if data is None:
             data = self.fetcher(0, self.size)
         self.data = data
 
-    def _fetch(self, start, end):
-        return self.data[start:end]
+    def _fetch(self, start: int | None, stop: int | None) -> bytes:
```

Comment:
I added support for `start = stop = None`, but hapyp to revert that and ust add a `# type: ignore` instead (complains that there's a different type signature than the base class).